lib/puppet/provider.rb

```diff
@@ -242,6 +242,8 @@ def initialize(name, path, confiner)
       @optional = false
       @confiner = confiner
       @custom_environment = {}
+      @command_resolver = Puppet::Util
+      @command_executor = Puppet::Util::Execution
```

Comment:
The `@command_resolver` and `@command_executor` are passed to the `Puppet::Provider::Command` instantiation on line 278.